### DISCUSSION TITLE
Live session for Data ingestion

### DISCUSSION THREAD

**Discussion by gprao08** [Sunbird-cQube/community#832] [General]
@bommalla @RNarmatha 
Could you please provide a live session for data ingestion, we have our teachers attendance data for the month of July & August. Once we familiar we can proceed further. Also we have missed PMPOSHAN data in the portal. Kindly help us to proceed further.

**Comment by bommalla** [comment]
@gprao08 Please tell me your availability then I can schedule the call accordingly.

**Reply by gprao08** [reply]
@bommalla Today & Tmrw available. Plz schedule and intimate before so that I will prepare accordingly.

**Comment by bommalla** [comment]
@gprao08 @RNarmatha Today at what your available please let me know then I can schedule a call accordingly.

**Comment by bommalla** [comment]
@gprao08 @RNarmatha , If you have school management data.Please prepare the school management dimension as well.Please find the below template and prepare school management data accordingly. 

https://docs.google.com/spreadsheets/d/1JYRRzuBswoHb8DlBF7o4UXHNXUmmf-L-0Ae3_zEWcSQ/edit#gid=2083524230

**Comment by RNarmatha** [comment]
@bommalla - Please  scehdule a call to support them for ingesting the teacher attendance

**Comment by bommalla** [comment]
@gprao08 @RNarmatha Please tell me your availability  then I can schedule call accordingly .

**Comment by bommalla** [comment]
@gprao08 @RNarmatha, Now cqube_minio bucket is created successfully.I got the 200 response from postman API but event file not ingested into cqube_minio bucket.I'm debugging on it once it is done then I'll update you.

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
I checked the teachers attendance data and found some mistakes.Please find the below points.

1. Each event file count is mismatch
2. Some of the event file having more records.
3. Some of the event file having less records.please make sure all three event files contain same records.
4. some duplicate and none values is there in the even files.
5. Please put date format should be dd/mm/yy format

Note: Once the above steps is done and share me the latest data to us.

**Reply by gprao08** [reply]
Hi Rajesh @bommalla 
We have completed the above steps and June month data ready now. Kindly schedule and intimate us. Tq

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Just now I checked the latest teacher_attendance data.Please change date format dd-mm-yy to dd/mm/yy.

**Reply by gprao08** [reply]
@bommalla @RNarmatha  Modified the date format from dd-mm-yy to dd/mm/yy

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

I tested the teacher_attendance data in my local machine.Data is showing in the UI dashboard but that data is wrong because of  data issue.I informed to AP team. Once data is ready please share to us.

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
We tested our side and the data errors as explained by you is give below:
1. Teachers Present marked is exceed the teachers marked data. 
2. Teachers marked data is exceeds the Total teachers data.
We have cross checked the data which we submitted to you and not found any errors, in case any errors identified plz give the details where the errors identified. So that we can join the call and resolve further.

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
June month data shared yesterday 14.11.23. Plz check and reply

**Comment by RNarmatha** [comment]
hi @gprao08 - we received the data.  we  received the data, yesterday is diwali hoilday. we  will get back to you. 

Thanks

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Please follow the below steps and delete the existing containers and re install the cqube again.
**step: 1**
sudo docker stop $(sudo docker ps -aq)
sudo docker rm $(sudo docker ps -aq)
sudo docker rmi $(sudo docker images -q)
sudo docker volume prune -y
sudo docker volume prune
sudo docker volume rm adapter_volume monitoring_grafana_data monitoring_prometheus_data python_executable_env
sudo docker network prune

sudo rm -rf /opt/cqube
sudo rm -rf /opt/containerd

sudo rm -rf ~/cqube-devops
**step:2**
Clone repo again by using below command
git clone https://github.com/Sunbird-cQube/cqube-devops.git
git checkout release v5.0.3
git pull
git branch
Then you can start the installation 
sudo ./install.sh

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Now AP server is up. please check the below credentials 
URL : https://vsk.ap.gov.in/
user_name : test-user	
password : Test@123

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Please follow the below ingestion document and do the ingestion.
https://docs.google.com/document/d/1eoElNSMm9g_IwmWBVFDmfRfYgRdalNxG/edit

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
While ingest new_programme there is an error showing in VSK_grammar (404 not found). Screen shot attached for your reference. Kindly help us to resolve.
![image](https://github.com/Sunbird-cQube/community/assets/143190553/bd4e3926-fdf2-4b35-9710-717a6f208e88)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Please check you database schema is there are not.if it's there please check the internet access in your local machine

Note:If you connected with VPN, at the time postman is not working please check and try to ingest.

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
Hi Rajesh, Could you plz give steps to check database schema, if not plz tell me how to do the task?

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Please follow the below steps to check the database schema.
1. Please run the below command in cqube-devops directory.
sudo docker exec -it postgres_app psql -U cqube_user -d cqube
2. After that you can type below commands in your terminal
\dt spec.*
select count(*) from spec."DimensionGrammar";
select count(*) from spec."EventGrammar";
3. If you want check the datasets in the database please run the below command
\dt datasets.*
select * from datasets."table_name";

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
As provided the commands above found database schema. Screen shots attached for your reference.
![screen1](https://github.com/Sunbird-cQube/community/assets/143190553/993b1178-39f5-4495-87a2-675b88a39151)
![screen2](https://github.com/Sunbird-cQube/community/assets/143190553/286778ed-8063-4492-b198-260eb969c2e8)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

what was the issue ? Can you please explain me in detail ?

**Reply by gprao08** [reply]
While trying to upload VSK_Grammer files after modify Programme_yml file in New_Programme it shows the error.

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
I scheduled call at 12 PM please be available at the time.

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Meeting details

AP ingestion issue
Monday, November 27 · 12:00 – 1:00pm
Time zone: Asia/Kolkata
Google Meet joining info
Video call link: https://meet.google.com/gmh-mhtg-dhr
Or dial: ‪(US) +1 636-336-8081‬ PIN: ‪580 612 438‬#
More phone numbers: https://tel.meet/gmh-mhtg-dhr?pin=9272630715330

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
As discussed I have upload all the 13 dimensions into Process_input foler. Now shall I run the below command?
**yarn cli ingest-data --filter='sch_att'**
Plz let me know where shall I run the above command. Is it $: /cQube_devops ?
![image](https://github.com/Sunbird-cQube/community/assets/143190553/e38f513e-4950-475a-aeaf-5245ed3a5c21)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Please go through the below mentioned path and run the command.
cd /cqube-devops
sudo docker exec -it nifi_app bash
cd /Sunbird-cQube/processing-ms/blob/dev/impl/c-qube/ingest/
then run the below command in the above location.
yarn cli ingest-data --filter='sch_att'

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
Tq. After successfully yarn cli command it displays successfully ingested all the files. But not data displayed in UI for the month of July 2023. Could you plz help me to resolve the issue

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Please check the below command and  send me the logs screenshot.
sudo docker logs ingest_app

**Reply by gprao08** [reply]
![screen1](https://github.com/Sunbird-cQube/community/assets/143190553/0dbd962d-37c1-4738-bc59-40a561a199ef)
![screen2](https://github.com/Sunbird-cQube/community/assets/143190553/e2efdf36-b4aa-4aee-bcba-895a80fbcc5c)
![screen3](https://github.com/Sunbird-cQube/community/assets/143190553/374a68b3-e9ae-4699-b17c-6c2b929ae1b9)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

I'll check and get back to you sir.Please give the server access for 10 min

**Reply by gprao08** [reply]
ok. Tq

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Just now I checked the database,updated data is not there in the database.please check the July data from your end and also send me

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
As we discussed yesterday please send me the latest teacher-attendance data.

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Just now I checked,data contain "null" land "none" values is there.Please check the data and ingest again.please find the screenshot for your reference.

![image](https://github.com/Sunbird-cQube/community/assets/86966482/3918ce38-66af-45d9-a85d-ca6110f687e2)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

I found one more mistake in the data, the column name in raw file having "statecategory_name" but in cqube template having "school_statecategory_name" please change it and ingest again.

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
Hi Rajesh
After I modify and processed again and run **yarn cli ingest-data --filter='sch_att'** it showed "cleaned all files" later no 'Teachers attendance present' in UI dashboard. Screen shot attached for your reference.
![yarn command](https://github.com/Sunbird-cQube/community/assets/143190553/598aafa2-084e-468a-9b7b-c380024cf9f0)

**Reply by gprao08** [reply]
![yarn cli](https://github.com/Sunbird-cQube/community/assets/143190553/0b36e642-1338-4cb4-b9af-f71b38f00e4a)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Still your facing issue please follow the below steps.

1.nuke the db in the server by using below command
yarn cli nuke-db
2. upload the vsk grammars and dimension grammars from the postman application
3. upload the all raw dimensions by using new_programs API 
4. upload the all three teacher-attendance event files by using new_programs API(first upload june month data if it's works fine then upload july month data) by using filter command
5. upload the all VSK raw file(pgi,nas,udise,diksha) by using national_programs API
6. After that schedule the "Run_adapters" and "Run Latest Code local" processor groups one by one.

**Reply by gprao08** [reply]
@bommalla Plz let me know where (the path) to run nuke-db command

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Please find the below path for nuke the db
sudo docker exec -it nifi_app bash
cd /Sunbird-cQube/processing-ms/blob/dev/impl/c-qube/ingest/

Please run yarn cli nuke-db command in above mentioned path

**Comment by gprao08** [comment]
@bommalla 
Could you plz tell me whare the dimensions & data files need to remove after I successfully run adapters. Plz find the below screen shot
![image](https://github.com/Sunbird-cQube/community/assets/143190553/f95a2a2a-ba02-46e4-ab5b-05376e0b6a53)

**Reply by bommalla** [reply]
@gprao08 @RNarmatha 
No need to delete any grammar files sir, after installation I deleted. you can go with the next step

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

July month data got ingested. Now it's up and running.Please find the attached screenshot for your reference.
![Screenshot from 2023-11-30 15-03-32](https://github.com/Sunbird-cQube/community/assets/86966482/9ef9e3f3-829c-4b97-823b-4a71036530a0)

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 

Separate two different months teacher-attendance data ingested in release v5.0.3, it will show latest month data only. if you want ingest teacher-attendance data monthly wise you can install release v5.0.5 version.
Note: Month wise data ingestion feature is not there in the cqube release v5.0.3 version.

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
Can we go for version 5.0.5, is it available in Git repository as explained by you earlier while training? In case upgrade version all data need to ingest again?

**Comment by bommalla** [comment]
@gprao08 @RNarmatha 
Yes,release v5.0.3 to v5.0.5 there is no up-gradation so we can install v5.0.5 freshly and do the ingestion again.
